### PR TITLE
Hot fix for WaveCalib crash

### PR DESCRIPTION
**Pull request by profxj**
As titled.  This was inspired by a GNIRS crash as reported
on the PypeIt Users Slack.

This only pushes the crash down into extraction as we have
not yet allowed for masked orders there.  Raising an Issue
on that.

Ignore the KOA Notebook that comes in.  That will be updated
in the `adap2020` branch.

### CODE REVIEW

File: proposals/ADAP/2020/KOA_Table.ipynb [hunk 1]
```diff
@@ -0,0 +1,1165 @@
+{
```
Comment (jhennawi) [inline proposals/ADAP/2020/KOA_Table.ipynb:1]: Please remove this file.

**Comment by profxj** [comment]
https://travis-ci.org/github/pypeit/PypeIt/builds/702085575

**Comment by profxj** [comment]
================ 229 passed, 605 warnings in 364.98s (0:06:04) =================

File: CHANGES.rst [hunk 2]
```diff
@@ -9,6 +9,7 @@
  - KCWI reduction complete up to spec2D frames
  - Implemented new flatfield DataContainer to separate
    pixelflat and illumflat
+ - Deal with wavecalib crash
```
Comment (kbwestfall) [inline CHANGES.rst]: This isn't technically part of 1.0.5 so I made a new category for the hotfixes and added this there.

File: pypeit/core/extract.py [hunk 3]
```diff
@@ -1814,7 +1814,7 @@ def ech_objfind(image, ivar, slitmask, slit_left, slit_righ, order_vec, maskslit
     # Find the spat IDs
     gdslit_spat = np.unique(slitmask[slitmask >= 0]).astype(int)  # Unique sorts
     if gdslit_spat.size != norders:
-        msgs.error("You have not dealt with masked orders properly")
+        msgs.error("PypeIt has not dealt with masked orders in echelle. Contact the Developers!")
```
Comment (kbwestfall) [inline pypeit/core/extract.py]: I made this message much longer.

File: pypeit/core/wavecal/autoid.py [hunk 4]
```diff
@@ -1259,11 +1258,14 @@ def report_final(self):
...
+            if slit in self.bad_slits:
+                msgs.warn(badmsg)
+                continue
             if self.all_patt_dict[str(slit)] is None:
                 msgs.warn(badmsg)
                 continue
             st = str(slit)
-            if len(self.wv_calib[st]) == 0:
+            if self.wv_calib[st] is None or len(self.wv_calib[st]) == 0:
                 print("Bad solution for slit: {}".format(st))
```
Comment (kbwestfall) [inline pypeit/core/wavecal/autoid.py]: Changed this to `msgs.warn` instead of print.

File: pypeit/core/wavecal/autoid.py [hunk 5]
```diff
@@ -1259,11 +1258,14 @@ def report_final(self):
             if slit not in self.ok_mask:
                 msgs.warn(badmsg)
                 continue
+            if slit in self.bad_slits:
```
Comment (kbwestfall) [inline pypeit/core/wavecal/autoid.py:1261]: Can't we combine all these if statements into a single one?

File: pypeit/slittrace.py [hunk 6]
```diff
@@ -308,6 +308,24 @@ def spatid_to_zero(self, spat_id):
         """
         return np.where(self.spat_id == spat_id)[0][0]
 
+    def slitord_to_zero(self, slitord):
```
Comment (kbwestfall) [inline pypeit/slittrace.py:311]: As far as I can tell, this is only used in a loop.  There's a way to do this in one go without the loop that should be more efficient, but not sure it's worth the effort.

Comment (profxj) [reply pypeit/slittrace.py:311]: I can imagine it being used outside of a loop.  
Leaving as is for now.

**Comment by profxj** [comment]
merging

**Pull request merged**